packages/agents-hosting/src/app/adaptiveCards/adaptiveCardsActions.ts

```diff
@@ -0,0 +1,293 @@
+   * @param verb - A string, RegExp, RouteSelector, or an array of these to match the action verb.
+   * @param handler - A function to handle the action submission.
+   * @returns The Teams application instance.
+   */
+  public actionSubmit<TData = Record<string, any>>(
+    verb: string | RegExp | RouteSelector | (string | RegExp | RouteSelector)[],
+    handler: (context: TurnContext, state: TState, data: TData) => Promise<void>
+  ): AgentApplication<TState> {
+    const filter = this._app.options.adaptiveCards?.actionSubmitFilter ?? DEFAULT_ACTION_SUBMIT_FILTER;
+    (Array.isArray(verb) ? verb : [verb]).forEach((v) => {
+      const selector = createActionSubmitSelector(v, filter)
+      this._app.addRoute(selector, async (context, state) => {
+        const a = context?.activity
+        if (a?.type !== ActivityTypes.Message || a?.text || typeof a?.value !== 'object') {
+          throw new Error(`Unexpected AdaptiveCards.actionSubmit() triggered for activity type: ${a?.type}`)
+        }
+
+        await handler(context, state as TState, (parseAdaptiveCardInvokeAction(a.value)) as TData ?? {} as TData)
+      })
+    })
+    return this._app
+  }
+
+  /**
+   * Registers a handler for the search event.
+   * @param dataset - A string, RegExp, RouteSelector, or an array of these to match the dataset.
+   * @param handler - A function to handle the search query.
+   * @returns The Teams application instance.
+   */
+  public search (
+    dataset: string | RegExp | RouteSelector | (string | RegExp | RouteSelector)[],
+    handler: (
+      context: TurnContext,
+      state: TState,
+      query: Query<AdaptiveCardsSearchParams>
+    ) => Promise<AdaptiveCardSearchResult[]>
+  ): AgentApplication<TState> {
+    (Array.isArray(dataset) ? dataset : [dataset]).forEach((ds) => {
+      const selector = createSearchSelector(ds)
+      this._app.addRoute(
+        selector,
+        async (context, state) => {
+          const a = context?.activity
+          if (a?.type !== 'invoke' || a?.name !== SEARCH_INVOKE_NAME) {
+            throw new Error(`Unexpected AdaptiveCards.search() triggered for activity type: ${a?.type}`)
+          }
+
+          const parsedQuery = parseValueSearchQuery(a.value)
+          const query: Query<AdaptiveCardsSearchParams> = {
+            count: parsedQuery.queryOptions?.top ?? 25,
+            skip: parsedQuery.queryOptions?.skip ?? 0,
+
+            parameters: {
+              queryText: parsedQuery.queryText ?? '',
+              dataset: parsedQuery.dataset ?? ''
+
+            }
+          }
+
+          const results = await handler(context, state, query)
+          if (!context.turnState.get(INVOKE_RESPONSE_KEY)) {
+            const response = {
+              type: AdaptiveCardInvokeResponseType.SEARCH,
+              value: {
+                results
+              }
+            }
+
+            await context.sendActivity({
+              value: { body: response, status: 200 } as InvokeResponse,
```

Comment:
Is there a way the user can select/override the status code here? I think depending on the response this won't be 200 all the time

Reply:
good catch. I'm following dotnet where we have the same pattern

https://github.com/microsoft/Agents-for-net/blob/139b4ca4f6195bcfa25ae085c04adfd3467fb669/src/libraries/Builder/Microsoft.Agents.Builder/App/AdaptiveCards/AdaptiveCard.cs?plain=1#L347C15-L355C19

@goldenwitch do you know how to override the `invokeResponse.status` from a search handler?

/cc @tracyboehrer

Reply:
I don't really know in this case.  There are times when Teams seems not to care, but this isn't always the case (for other Invokes).

Reply:
I was trying to learn about AdaptiveCards search invokes by using this sample:

https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/bot-type-ahead-search-adaptive-cards

note that the user code is responsible to set the invoke status:

https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/e4a9849e9895d026e80f24601f50c44c0a55e6ae/samples/bot-type-ahead-search-adaptive-cards/nodejs/bots/teamsBot.js#L342-L367

however, the same sample in TeamsAI does not offer that capability.

https://github.com/microsoft/teams-ai/tree/main/js/samples/02.teams-features/b.adaptiveCards.typeAheadBot

/c @Stevenic